osu.Game/Online/API/Requests/DownloadBeatmapSetRequest.cs

```diff
@@ -2,28 +2,21 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using osu.Game.Beatmaps;
-using System;
 
 namespace osu.Game.Online.API.Requests
 {
-    public class DownloadBeatmapSetRequest : APIDownloadRequest
+    public class DownloadBeatmapSetRequest : ArchiveDownloadRequest<BeatmapSetInfo>
     {
-        public readonly BeatmapSetInfo BeatmapSet;
-
-        public float Progress;
-
-        public event Action<float> DownloadProgressed;
-
         private readonly bool noVideo;
+        private readonly BeatmapSetInfo set;
 
         public DownloadBeatmapSetRequest(BeatmapSetInfo set, bool noVideo)
+            : base(set)
         {
             this.noVideo = noVideo;
-            BeatmapSet = set;
-
-            Progressed += (current, total) => DownloadProgressed?.Invoke(Progress = (float)current / total);
+            this.set = set;
```

Comment:
just noticed this unnecessary local variable. `Model` is already on parent class.

Reply:
feel free to fix if you are able to. i'll be doing another pass on this later on failing that.